scripts/collect_activations.py

```diff
@@ -1,77 +1,169 @@
 #!/usr/bin/env python3
 
 import argparse
+import os
 import pickle
+from typing import Literal
 
 import torch
 from tqdm.auto import tqdm
-from transformers import AutoModelForCausalLM
+from transformers import AutoModelForCausalLM, AutoTokenizer
 
-from cot_probing.activations import (
-    Activations,
-    QuestionActivations,
-    clean_run_with_cache,
-)
+from cot_probing.activations import clean_run_with_cache
+from cot_probing.eval import EvalQuestion
+from cot_probing.task import INSTRUCTION_STR
 from cot_probing.typing import *
 
+LocTypeToCache = Literal["resp", "instr", "quest", "fsp"]
+ActivationsInLayer = Float[torch.Tensor, "pos d_model"]
+
 
 def parse_arguments():
+    # Example usage:
+    # python scripts/collect_activations.py --eval-questions /workspace/cot-probing-hf/google--gemma-2-2b/movie_recommendation/bias-always-A_seed-0_total-10/eval_questions.pkl
+
     parser = argparse.ArgumentParser()
     parser.add_argument(
-        "eval_results_path",
+        "--eval-questions",
+        "-f",
         type=str,
-        help="Path to the evaluation results pickle file",
+        default=None,
+        help="Absolute path to eval_questions.pkl file",
     )
     return parser.parse_args()
 
 
-if __name__ == "__main__":
-
-    args = parse_arguments()
-    eval_results_path = args.eval_results_path
-
-    with open(eval_results_path, "rb") as f:
-        eval_results = pickle.load(f)
-
-    model_name = eval_results.model_name
-    task_name = eval_results.task_name
-    seed = eval_results.seed
-    num_samples = eval_results.num_samples
-    questions = eval_results.questions
+def process_questions(
+    output_folder: str,
+    biased: bool,
+    model: AutoModelForCausalLM,
+    tokenizer: AutoTokenizer,
+    eval_questions: list[EvalQuestion],
+    loc_types_to_cache: set[LocTypeToCache] = {"resp"},
+):
+    if biased:
+        output_folder = os.path.join(output_folder, "biased_context")
+    else:
+        output_folder = os.path.join(output_folder, "unbiased_context")
+
+    # Load few-shot prompts
+    fsps_path = os.path.join(output_folder, "tokenized_fsps.pkl")
+    with open(fsps_path, "rb") as f:
+        fsps = pickle.load(f)
+
+    # Load tokenized responses
+    tokenized_responses_path = os.path.join(output_folder, "tokenized_responses.pkl")
+    with open(tokenized_responses_path, "rb") as f:
+        tokenized_responses = pickle.load(f)
 
-    print(
-        f"Processing {model_name} on {task_name} with seed {seed} and {num_samples} samples"
-    )
+    layers_to_cache = list(range(model.config.num_hidden_layers))
```

Comment:
if we're hardcoding it, then let's not pretend later in the code that we're not - it'll make it simpler